exasol/analytics/schema/column.py

```diff
@@ -50,3 +53,33 @@ def varchar_column(
             characterSet=characterSet,
         ),
     )
+
+
+def hashtype_column(
+    name: str,
+    bytes: Optional[int] = None,
+    bits: Optional[int] = None,
+) -> Column:
+    def size_and_unit() -> tuple[int, SizeUnit]:
```

Comment:
What was the point in making it a function?

Reply:
* First of all, this is in line with the other functions, eg. `varchar_column()`, etc.
* Second: It does not only create a column type but also a column, with a name _and_ the type. 
* Third: The interface is not `bytes` and `bit` rather than `size` and `unit`
* Fourth: If you refer to the error checks, then, yes, I was tempted to move these to class `ColumnType`.

But moving the error checks to class `ColumnType`, would require more changes
* switch from `dataclass` to ordinary class 
* add `__init__`
* implement `__eq__` and `__repr__`

In the end, I was reluctant if I should add these changes to the current PR. 
What is your opinion?

Reply:
Not a big deal of course. You usually create a function within a function if you need to compute something repeatedly, e.g. in a cycle or list comprehension. If you want to improve code readability you would move a piece of code into a separate function. To me, an inner function called only once looks a bit odd.

Reply:
I wanted to separate the error handling and different values to be returned from the part creating the actual instance of `Column`. But looking at it, you maybe right. 
See next push removing the function.